### DISCUSSION TITLE
No hardware acceleration / Vulkan fails on Fedora 40

### DISCUSSION THREAD

**Discussion by JKjakay** [86Box/86Box#4662] [Q&A] [answered]
### What happened?

Trying out the new 86Box 4.2 and I like it very much, except for one issue: I am unable to run this with hardware-accelerated QT or OpenGL. It will only work if I set the Renderer to "QT Software". All other options slow 86Box down to a crawl. It seems to be a Vulkan / Nvidia / Wayland driver issue.

Starting the AppImage in the shell shows this:
```
Qt: version 5.15.2, platform "wayland"
libpng warning: IDAT: Extra compressed data
Fontconfig warning: "/usr/share/fontconfig/conf.avail/05-reset-dirs-sample.conf", line 6: unknown element "reset-dirs"
Failed to load vulkan: Cannot load library vulkan: (vulkan: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden)
initInstance: No Vulkan library available
Failed to create platform Vulkan instance
```
The part "vulkan: Kann die Shared-Object-Datei nicht öffnen: Datei oder Verzeichnis nicht gefunden" translates as: "vulkan: cannot open shared object file: file or folder not found."

I have a Nvidia 4080 graphics card with newest 555 drivers installed from the rpmfusion repos, and running KDE Plasma (Wayland). Other graphics-related stuff works as it should, at least most games and apps do - with a few exceptions, of which this may be one.

### Configuration file

```ini
[General]
vid_renderer = qt_software
host_cpu = AMD Ryzen 7 7700X 8-Core Processor
emu_build_num = 6000
uuid = ae985417-4310-5dc9-841f-afb9ffe281ed

[Machine]
machine = ibmpc
cpu_family = 8088
cpu_speed = 4772728
cpu_multi = 1
cpu_use_dynarec = 0
fpu_softfloat = 0
time_sync = local
mem_size = 64

[Video]
gfxcard = cga

[Input devices]
mouse_type = none

[Sound]
fm_driver = nuked

[Network]
net_01_link = 0
net_02_link = 0
net_03_link = 0
net_04_link = 0

[Storage controllers]
fdc = none
hdc = none
cassette_enabled = 1
cassette_mode = load
```

### Operating system

Fedora 40

### CPU

AMD Ryzen 7 7700 X

### 86Box version

v4.2 Build 6000

### Build architecture

Linux - x64 (64-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Download source

Official website (Jenkins, GitHub)

**Comment by OBattler** [comment, marked as answer]
You are missing the libraries:

Depending on your setup you need:

```
dnf install vulkan-headers vulkan-loader-devel
```

or

```
apt-get install libvulkan-dev
```

or

```
pacman -S vulkan-headers vulkan-icd-loader
```

Also, converting this to discussion because it's not an 86Box bug.

**Comment by JKjakay** [comment]
Yes, that did the trick, thank you!
